packages/flutter/lib/src/material/refresh_indicator.dart

```diff
@@ -214,6 +253,14 @@ class RefreshIndicator extends StatefulWidget {
   /// [Future] must complete when the refresh operation is finished.
   final RefreshCallback onRefresh;
 
+  /// An optional callback function used to get current status of
+  /// [RefreshIndicator] to update UI as required, while using
+  /// [RefreshIndicator.noSpinner].
+  /// This is an optional parameter, used to fine tune app cases.
+  ///
+  /// Returns value of type: [RefreshIndicatorStatus]
+  final void Function(RefreshIndicatorStatus? status)? onStatusChange;
```

Comment:
Can `ValueChanged<RefreshIndicatorStatus?>?` be used as the parameter type here instead?